### ISSUE TITLE
LayoutView - ItemViews - reset calls empty

### ISSUE THREAD

**Issue by laszlogmenyhart** [marionettejs/backbone.marionette#2739]
Hi,

We found that Marionett.Region.reset function calls the empty function without parameter
(see: https://github.com/marionettejs/backbone.marionette/blob/master/lib/core/backbone.marionette.js line 862) in version 2.4.2. So the options is undefined in line 803.

Please check whether our bugfix is good or not!
Our solution is this from line 800:

 empty: function(options) {
      var view = this.currentView;

     // If there is no view in the region
      // we should not remove anything
      if (!view) { return; }
      if (!options) {
         options = {};
      }
      options = _.extend(options, view.options);
      var preventDestroy = Marionette._getValue(options, 'preventDestroy', this);

//  Original code from here
      view.off('destroy', this.empty, this);
      this.triggerMethod('before:empty', view);
      if (!preventDestroy) {
        this._destroyView();
      }
      this.triggerMethod('empty', view);
      // Remove region pointer to the currentView
      delete this.currentView;
      if (preventDestroy) {
        this.$el.contents().detach();
      }
      return this;
    },

Thanks!

**Comment by paulfalgout** [comment]
I believe this is fixed in the `2.4.3` release which will be coming out in the next day or so:
https://github.com/ianmstew/backbone.marionette/commit/0aec7d4f9579486f367e1e93a1c999bb8474e94a

**Comment by laszlogmenyhart** [comment]
OK Thanks!
I will test and come back if this is not enough for us.

**Comment by paulfalgout** [comment]
`2.4.3` is up.

**Comment by laszlogmenyhart** [comment]
Hi!

It seems that we have the issue yet.

Our solution was not good, because we added the preventDestroy property to the view, but now we can see that it can be only an element of the options (second args) of show function.

Now we modified our code but we found that it is not used at calling other functions or it is not handled correctly or we still have another issue.

Would you be so kind to please check it in Marionett.region.show function?

1) We have a CompositeView ("list")
2) from this we step to another LayoutView ("edit") with some Views and CompositeViews ("items")
3) Users can be go back to the prevoius view just to check something
4) and go forward to the same LayoutView (continue the edition)
5) when we show this LayoutView the original "items" are destroyed even if we set the preventDestroy option at calling the show "edit" LayoutView.
_shouldDestroyView = false
_shouldShowView = true
So this._renderView(view); is called (I think it would be not needed here!), 
view.render() calls this._reInitializeRegions(); which resets the regions and every "items" will be destroyed because "options" parameter is not set.

Thanks,
LAci

**Comment by paulfalgout** [comment]
I think a gist or a fiddle might be helpful here.  But what is the use-case for the `preventDestroy`?   In general it is much better to just destroy and rebuild views when you need them.  `preventDestroy` is very much an edge-case feature.

**Comment by paulfalgout** [comment]
This will be fixed here https://github.com/marionettejs/backbone.marionette/pull/2850
